EasyCommands.Tests/ScriptTests/ScriptTest.cs

```diff
@@ -0,0 +1,182 @@
+        /// invoke a callback that will attempt to retrieve the stored blocks.
+        /// </summary>
+        /// <typeparam name="T">The type of the blocks we're mocking retrieval for.</typeparam>
+        /// <param name="name">The custom name of the block(s) we're mocking.</param>
+        /// <param name="blockMocks">Blocked mocks to be stored and used later.</param>
+        public void MockBlocksOfType<T>(String name, params Mock<T>[] blockMocks) where T : class, IMyTerminalBlock
+        {
+            SetupMockBlocksByType(name, blockMocks);
+
+            // Set up the mocked GridTerminalSystem to invoke the appropriate callback method for the appropriate type
+            gridMock.Setup(g => g.GetBlocksOfType(It.IsAny<List<T>>(), It.IsAny<Func<T, bool>>()))
+                .Callback((Action<List<T>, Func<T, bool>>)GetBlocksOfTypeCallback);
+        }
+
+        /// <summary>
+        /// Given a group name, set up a mock group that will be retrieved by the GridTerminalSystem.
+        /// Store the given mocked blocks for later retrieval.
+        /// 
+        /// Also, set up the group's GetBlocksOfType method to use a Moq Callback so that
+        /// those mocked blocks can be found later.
+        /// </summary>
+        /// <typeparam name="T">The type of the blocks being mocked for this group.</typeparam>
+        /// <param name="groupName">The name of the group being mocked.</param>
+        /// <param name="blockMocks">The blocks being mocked and that will be returned later.</param>
+        public void MockBlocksInGroup<T>(String groupName, params Mock<T>[] blockMocks) where T : class, IMyTerminalBlock
+        {
+            var mockGroup = new Mock<IMyBlockGroup>();
+            mockGroup.Setup(g => g.Name).Returns(groupName);
+            mockGroup.Setup(g => g.GetBlocksOfType(It.IsAny<List<T>>(), It.IsAny<Func<T, bool>>()))
+                .Callback((Action<List<T>, Func<T, bool>>)GetBlocksOfTypeCallback);
+            mockedGroups.Add(mockGroup.Object);
+
+            SetupMockBlocksByType(blockMocks);
+        }
+
+        private void GetBlocksOfTypeCallback<T>(List<T> blocks, Func<T, bool> collect) where T : class, IMyTerminalBlock
+        {
+            // Find the appropriate blocks by type and then invoke the given collector function on them as a filter
+            blocks.AddRange(mockedBlocksByName.GetValue<T>(typeof(T))
+                .Select(x => x)
+                .Where(collect != null ? collect : (x) => true)
+                .ToList());
+        }
+
+        private void GetBlockGroupsCallback(List<IMyBlockGroup> groups, Func<IMyBlockGroup, bool> collect)
+        {
+            // Invoke the given collector function on all groups and return those groups matching
+            groups.AddRange(mockedGroups
+                .Select(x => x)
+                .Where(collect != null ? collect : (x) => true)
+                .ToList());
+        }
+
+        public void Dispose()
```

Comment:
Do we need this one?

nit: I think default is LogLevel.INFO.  I think this also hints at doing the defaulting in the Program constructor, and then overriding in the our test setup after creating the program. 

Thoughts?

Reply:
Oh for sure, these should all be instance variables. The Dispose will be useful for anything we can't change to an instance variable, or any other post-execution cleanup we want to do. FYI, the `Dispose` method is invoked automatically when we close out the `using` statement.

> nit: I think default is LogLevel.INFO.

Yup! That's why I'm setting it to that in Dispose.

Anyway, you're totally right on this better being done with instance variables, but I didn't want to make a bunch of changes to your state management variables in this CR. I can do that in a follow-up, though!